spec/openhab/dsl/items/builder_spec.rb

```diff
@@ -295,13 +295,47 @@
       expect(StringItem1.thing).to be things["astro:sun:home"]
     end
 
+    it "can link an item to multiple channels" do
+      things.build do
+        thing "astro:moon:home", "Astro Moon Data", config: { "geolocation" => "0,0" }
+      end
+      items.build do
+        date_time_item "DateTime1" do
+          channel  "astro:sun:home:rise#start"
+          channel  "astro:moon:home:rise#start"
+        end
+      end
+      expect(DateTime1.things).to match_array([things["astro:sun:home"], things["astro:moon:home"]])
+    end
+
     it "can link to an item channel with a profile" do
       items.build do
         date_time_item "LastUpdated", channel: ["astro:sun:home:season#name", { profile: "system:timestamp-update" }]
       end
       expect(LastUpdated.thing).to be things["astro:sun:home"]
     end
 
+    it "combines thing (string) and channel" do
+      items.build do
+        string_item "StringItem1", thing: "astro:sun:home", channel: "season#name"
+      end
+      expect(StringItem1.thing).to be things["astro:sun:home"]
+    end
+
+    it "combines thing and channel" do
+      items.build do
+        string_item "StringItem1", thing: things["astro:sun:home"], channel: "season#name"
+      end
+      expect(StringItem1.thing).to be things["astro:sun:home"]
+    end
+
+    it "ignores thing when channel contains multiple segments" do
+      items.build do
+        string_item "StringItem1", thing: "foo:baz:bar", channel: "astro:sun:home:season#name"
```

Comment:
Should we log a warning in this scenario?

Reply:
no, I don't think so, because it's valid to provide thing/channel for one linked channel, and a fully qualified channel for another.